core/src/main/java/io/ddf/DDFCache.java

```diff
@@ -13,47 +17,62 @@
 import org.apache.commons.lang.exception.ExceptionUtils;
 
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
 
 /**
  */
 public class DDFCache extends ALoggable {
 
-  private LoadingCache<UUID, DDF> mDDFCache;
+  Striped<Lock> uuidLocks = Striped.lock(Runtime.getRuntime().availableProcessors() * 4);
+
+  /**
+   * Keeping DDF without name
+   */
+  private Cache<UUID, DDF> mDDFCache;
+
+  /**
+   * Keeping DDF that has name in a separate cache, so eviction on DDF without name
+   * does not affect DDF with name
+   */
+  private Cache<UUID, DDF> mDDFwithNameCache;
```

Comment:
Can we have a short paragraph that describes why we have 2 caches here, together with a `TODO` note asking for future improvements?